### ISSUE TITLE
Improve labelling of elements in Gallery

### ISSUE THREAD

**Issue by jasmussen** [WordPress/gutenberg#11396]
From https://github.com/WordPress/gutenberg/pull/11205#pullrequestreview-170864593, the gallery block is now keyboard accessible, but we can improve the way elements are announced to screen readers:

![screenshot](https://user-images.githubusercontent.com/90871/47877121-07b75380-de13-11e8-8680-678e8610edb0.png)

It might be good to have an aria-label or something on images with no alt attributes… but even then it might be nice to preface it with "image 2 of 3: $alt"

**Comment by BE-Webdesign** [comment]
I can take this on, thank you for bringing it up!
